toolbox/gpu-operator/must-gather.sh

```diff
@@ -0,0 +1,78 @@
+#! /bin/bash -ex
+
+if [ "$0" == "/usr/bin/gather" ]; then
+    echo "Running as must-gather plugin image"
+    export ARTIFACT_DIR=/must-gather
+
+    THIS_DIR="$(dirname "$(readlink -f "$0")")"
+else
+    # avoid sourcing _common.sh and messing up different env variables
+    if [ -z "${ARTIFACT_DIR:-}" ]; then
+        export ARTIFACT_DIR="/tmp/ci-artifacts_$(date +%Y%m%d)"
+        echo "Using '$ARTIFACT_DIR' to store the test artifacts (default value for ARTIFACT_DIR)."
+    else
+        echo "Using '$ARTIFACT_DIR' to store the test artifacts."
+    fi
+
+    export ARTIFACT_DIR="$ARTIFACT_DIR/$(date +%H%M%S)__gpu-operator__must-gather"
```

Comment:
just following here the same convention as in [`_comon.sh`](https://github.com/openshift-psap/ci-artifacts/blob/master/toolbox/_common.sh#L29), but avoid to source it as it Ansible log directories env var messed everything up.

the `__` separate the `<ts>__<dirname/toolname>__<operation name>` in the artifacts:
```
234108__cluster__capture_environment/
234112__entitlement__test/
234136__entitlement__deploy/
234138__entitlement__wait/
234526__nfd__has_nfd_labels/
```

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/periodic-ci-openshift-psap-ci-artifacts-release-4.7-gpu-operator-e2e-master/1386817554084794368/artifacts/gpu-operator-e2e-master/nightly/artifacts

Actually, I'm considering changing the `<ts>` to a simpler `<idx>`, computed by counting the number of files in the artifact directory, WDYT? (and @ArangoGutierrez / @dagrayvid ?) it would give this:

```
000__cluster__capture_environment/
001__entitlement__test/
002__entitlement__deploy/
003__entitlement__wait/
004__nfd__has_nfd_labels/
```

and resolve the midnight problem

```
000711__gpu-operator__run_gpu_burn/
001419__gpu-operator__capture_deployment_state/
234108__cluster__capture_environment/
234112__entitlement__test/
```